app/README.md

```diff
@@ -0,0 +1,221 @@
+[Onet](../README.md) ::
+App support
+
+# App support
+
+This is a collection of libraries to support CLI-apps that work with the onet
+framework. The go-library simplifies loading and saving of group-configurations
+and supports some very basic input/output functions.
+
+For integration support to test the CLI-apps, `libtest.sh` supports a range
+of tests to make sure that the binary behaves as needed.
+
+# LibTest.sh
+
+This is a specialized bash-library to handle the following parts of the test:
+
+* compiling of a conode with the required services
+* setting up and removing the directory structure to run the tests
+* different test primitives to test failure or success of running the binary
+
+One goal of `libtest.sh` is to test the binary in a way as close to what a user
+would do when running the binary. This leads to a better understanding whether
+the binary has a logical setup of the commands and command line options.
+
+## Command Line Options
+
+The default for `libtest.sh`-enabled test is to:
+
+* compile the binary
+* run all the tests in a temporary directory
+* delete the temporary directory after the test.
+
+Two command line options are available to infulence how the test is run:
+
+* `-nt` - NoTemp - sets up the test-environment in `./build` and doesn't delete
+it after the test is done. This is useful if you want to inspect the environment
+after the test run. *CAREFUL* because it will *NOT* recompile the binary. You
+will use this flag mostly while developing the `test.sh` file, so that you can
+quickly test if your new tests are correct or not.
+* `-b` - Build - only useful with `-nt` to force a build of the binary. While
+developing the tests, in a first time you might want to use `-nt` until your
+`test.sh` file is OK, then you will work on your go-code. To recompile the go-code,
+use this flag. Once the go-code is OK, you can remove this flag again and work
+on the `test.sh` file.
+
+## Test.sh setup
+
+The common name for using `libtest.sh` is a file called `test.sh` in the
+same directory as the CLI-binary. The usual structure for this file is:
+
+* Global variables and inclusion of `libtest.sh`
+* main-method setting up the environment and starting the tests
+* a list of test-methods
+* helper methods
+* a call to main, so that the main method is at the top and all necessary
+methods are defined before main is called.
+
+### Global Variables and inclusion of `libtest.sh`
+
+| Variable | Default | Explanation |
+| -------- | ------- | ----------- |
+| `DBG_TEST` | `0` | Show the output of the commands: 0=none, 1=test-names, 2=all |
+| `DBG_SRV` | `0` | DBG-level for server, passed as `-debug` to the `conode` command |
+| `NBR` | `3` | Highest number of servers and clients |
+| `NBR_SERVERS` | `NBR` | How many servers to configure |
+| `NBR_SERVERS_GROUP` | `NBR_SERVERS - 1` | How many servers to write to the
+group-file - per default, keep one server out of the group, to test what
```

Comment:
This doesn't format correctly, prob needs to be one giant line. Same for testGrep, testNGRep, testCount, etc.